test/gateway/pollingInterval.js

```diff
@@ -0,0 +1,797 @@
+    name: 'John',
+    lastName: 'Doe'
+  }
+
+  const resolvers = {
+    Query: {
+      me: (root, args, context, info) => user
+    },
+    Mutation: {
+      triggerUser: async (root, args, { pubsub }) => {
+        await pubsub.publish({
+          topic: 'UPDATED.USER',
+          payload: {
+            updatedUser: user
+          }
+        })
+
+        return true
+      }
+    },
+    Subscription: {
+      updatedUser: {
+        subscribe: async (root, args, { pubsub }) =>
+          pubsub.subscribe('UPDATED.USER')
+      }
+    },
+    User: {
+      __resolveReference: (user, args, context, info) => user
+    }
+  }
+
+  const userService = Fastify()
+  const gateway = Fastify()
+
+  t.tearDown(() => {
+    userService.close()
+    gateway.close()
+  })
+
+  userService.register(GQL, {
+    schema: `
+      extend type Query {
+        me: User
+      }
+
+      extend type Subscription {
+        updatedUser: User
+      }
+
+      extend type Mutation {
+        triggerUser: Boolean
+      }
+
+      type User @key(fields: "id") {
+        id: ID!
+        name: String!
+      }
+    `,
+    resolvers: resolvers,
+    federationMetadata: true,
+    subscription: true
+  })
+
+  await userService.listen(0)
+
+  const userServicePort = userService.server.address().port
+
+  gateway.register(GQL, {
+    subscription: true,
+    gateway: {
+      services: [
+        {
+          name: 'user',
+          url: `http://localhost:${userServicePort}/graphql`,
+          wsUrl: `ws://localhost:${userServicePort}/graphql`
+        }
+      ],
+      pollingInterval: 2000
+    }
+  })
+
+  await gateway.listen(0)
+
+  const ws = new WebSocket(
+    `ws://localhost:${gateway.server.address().port}/graphql`,
+    'graphql-ws'
+  )
+
+  t.equal(ws.readyState, 0)
+
+  const client = WebSocket.createWebSocketStream(ws, {
+    encoding: 'utf8',
+    objectMode: true
+  })
+
+  t.tearDown(() => {
+    client.destroy()
+  })
+
+  client.setEncoding('utf8')
+
+  client.write(
+    JSON.stringify({
+      type: 'connection_init'
+    })
+  )
+
+  client.write(
+    JSON.stringify({
+      id: 1,
+      type: 'start',
+      payload: {
+        query: `
+          subscription {
+            updatedUser {
+              id
+              name
+            }
+          }
+        `
+      }
+    })
+  )
+
+  client.on('data', (chunk) => {
+    const data = JSON.parse(chunk)
+
+    if (data.type === 'connection_ack') {
+      gateway.inject({
+        method: 'POST',
+        url: '/graphql',
+        body: {
+          query: `
+          mutation {
+            triggerUser
+          }
+        `
+        }
+      })
+    } else if (data.type === 'data' && data.id === 1) {
+      const { payload: { data: { updatedUser = {} } = {} } = {} } = data
+
+      t.deepEqual(updatedUser, {
+        id: 'u1',
+        name: 'John'
+      })
+
+      eventEmitter.emit('client.on.data')
+    }
+  })
+
+  await once(eventEmitter, 'client.on.data')
```

Comment:
There is no need for a custom eventEmitter. Just do `once(client, 'data')`

Reply:
I have replaced custom eventEmitter with client, but in some cases, I needed to trigger it twice (connection and action)